posthog/tasks/exports/video_exporter.py

```diff
@@ -0,0 +1,224 @@
+                        "+faststart",
+                        "-f",
+                        "mp4",
+                    ]
+                    if video_filter:
+                        cmd.extend(["-vf", video_filter])
+                    cmd.append(image_path)
+                    try:
+                        subprocess.run(cmd, check=True, capture_output=True, text=True)
+                    except subprocess.CalledProcessError as e:
+                        error_msg = f"ffmpeg failed with exit code {e.returncode}"
+                        if e.stderr:
+                            error_msg += f": {e.stderr.strip()}"
+                        raise RuntimeError(error_msg) from e
+                elif ext == ".gif":
+                    vf_parts = ["fps=12"]
+                    if measured_width is not None:
+                        vf_parts.append(f"scale={measured_width}:-2:flags=lanczos")
+                    vf = ",".join(vf_parts)
+                    try:
+                        subprocess.run(
+                            [
+                                "ffmpeg",
+                                "-hide_banner",
+                                "-loglevel",
+                                "error",
+                                "-y",
+                                "-ss",
+                                f"{pre_roll:.2f}",
+                                "-t",
+                                f"{float(recording_duration):.2f}",
+                                "-i",
+                                tmp_webm,
+                                "-vf",
+                                f"{vf},split[s0][s1];[s0]palettegen=stats_mode=single[p];[s1][p]paletteuse=dither=bayer:bayer_scale=5:diff_mode=rectangle",
+                                "-loop",
+                                "0",
+                                "-f",
+                                "gif",
+                                image_path,
+                            ],
+                            check=True,
+                        )
+                    except subprocess.CalledProcessError as e:
+                        error_msg = f"ffmpeg failed with exit code {e.returncode}"
+                        if e.stderr:
+                            error_msg += f": {e.stderr.strip()}"
+                        raise RuntimeError(error_msg) from e
+                else:
+                    shutil.move(tmp_webm, image_path)
+            finally:
+                try:
+                    context.close()
+                    browser.close()
+                except Exception:
+                    pass
+    except Exception as e:
+        with posthoganalytics.new_context():
+            posthoganalytics.tag("url_to_render", url_to_render)
+            posthoganalytics.tag("video_target_path", image_path)
+            capture_exception(e)
+        raise
+    finally:
+        if temp_dir_ctx:
+            try:
+                temp_dir_ctx.cleanup()
```

Comment:
If you instead use `temp_dir_ctx.cleanup(ignore_cleanup_errors=True)` then you don't need the try/except+pass construct at all here.